vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -110,11 +110,78 @@ def load_hosts
         end
       end
 
+      def parse_lscpu(lscpu_data)
+        parsed_data = {}
+
+        lscpu_data.each_line do |line|
+          parts = line.split(/:/)
+          value = parts[1].try(:chomp).try(:strip)
+          case parts[0]
+          when 'Socket(s)'
+            parsed_data['numvcpus'] = value
+          when 'Core(s) per socket'
+            parsed_data['cores_per_socket'] = value
+          when "CPU MHz"
+            parsed_data['cpu_speed'] = value
+          when "Model name"
+            parsed_data['cpu_type'] = value
```

Comment:
This might read nicer with each option on one line.

```ruby
          case parts[0]
          when 'Socket(s)'          then parsed_data['numvcpus'] = value
          when 'Core(s) per socket' then parsed_data['cores_per_socket'] = value
          when "CPU MHz"            then parsed_data['cpu_speed'] = value
          when "Model name"         then parsed_data['cpu_type'] = value
          end
```

Or ever better, factor out the setting from the key selection, to DRY it up...

```ruby
          key = case parts[0]
                when 'Socket(s)'          then 'numvcpus'
                when 'Core(s) per socket' then 'cores_per_socket'
                when "CPU MHz"            then 'cpu_speed'
                when "Model name"         then 'cpu_type'
                end
          parsed_data[key] = value
```